server/src/main/java/org/apache/cassandra/sidecar/modules/CdcModule.java

```diff
@@ -63,6 +69,17 @@ PeriodicTask sidecarPeerHealthMonitorTask(SidecarPeerHealthMonitorTask task)
         return task;
     }
 
+    @ProvidesIntoMap
+    @KeyClassMapKey(PeriodicTaskMapKeys.CdcRawDirectorySpaceCleanerTaskKey.class)
+    PeriodicTask cdcRawDirectorySpaceCleanercPeriodicTask(TimeProvider timeProvider,
+                                                          SystemViewsDatabaseAccessor systemViewsDatabaseAccessor,
+                                                          CdcConfiguration cdcConfiguration,
+                                                          InstanceMetadata instanceMetadata,
+                                                          SidecarMetrics metrics)
+    {
+        return new CdcRawDirectorySpaceCleaner(timeProvider, systemViewsDatabaseAccessor, cdcConfiguration, instanceMetadata, metrics);
+    }
```

Comment:
It is probably like a hack. But since `CdcRawDirectorySpaceCleaner` is annotated with `@Singleton` and `@Inject`, we can simply wire the singleton to the mapBinder. It is actually necessary to change to avoid breaking the `singleton` requirement, as `new`ing instance does not honor the annotations. 

```suggestion
    @ProvidesIntoMap
    @KeyClassMapKey(PeriodicTaskMapKeys.CdcRawDirectorySpaceCleanerTaskKey.class)
    PeriodicTask cdcRawDirectorySpaceCleanercPeriodicTask(CdcRawDirectorySpaceCleaner cleanerTask)
    {
        return cleanerTask;
    }
```